--- Dockerfile
+++ Dockerfile
@@ -6,14 +6,8 @@
 WORKDIR /app

 # Install dependencies based on the preferred package manager
-COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
-RUN \
-  if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
-  elif [ -f package-lock.json ]; then npm ci; \
-  elif [ -f pnpm-lock.yaml ]; then corepack enable pnpm && pnpm i --frozen-lockfile; \
-  else echo "Lockfile not found." && npm install; \
-  fi
+COPY package.json ./
+RUN corepack enable pnpm && pnpm install

 # Rebuild the source code only when needed
 FROM base AS builder
